src/main/java/com/microsoft/cse/helium/app/dao/GenresDao.java

```diff
@@ -0,0 +1,52 @@
+package com.microsoft.cse.helium.app.dao;
+
+import com.azure.data.cosmos.CosmosClient;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.spring.data.cosmosdb.core.convert.ObjectMapperFactory;
+import com.microsoft.cse.helium.app.models.Genre;
+import com.microsoft.cse.helium.app.services.configuration.IConfigurationService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Service
+public class GenresDao extends BaseCosmosDbDao {
+  private static final Logger logger = LoggerFactory.getLogger(GenresDao.class);
+
+  private static String genreQuery =
+      "select m.genre from m where m.type = 'Genre' order by m.genre ";
+
+  public GenresDao(IConfigurationService configService) {
+    super(configService);
+  }
+
+  /**
+   * getActors.
+   *
+   * @return
+   */
+  public Mono<List<String>> getGenres() {
+    logger.info("genreQuery " + genreQuery);
+
```

Comment:
Same here, I think it should be Flux<>.

Reply:
This is a workaround as CosmosDB SDK do not support value queries. To handle that scenario we need to convert Object Array to [{"genre":"action","genre":adventure"...}] to String Array from a stream ["action",adventure"...] .  CollectMap is the only way to handle stream of strings. I verified this with cosmos db team as well. We can move to return Flux only the "value" queries are supported by CosmosDB SDK